controllers/ie_controller_test.go

```diff
@@ -0,0 +1,177 @@
+package controllers
```

Comment:
Agree w/ Sam.  Keep the test in the same package.  It's the "Go way" -- even if it does feel wrong.  When Go compiles the source it will not compile test classes into the production binary, so it is safe.